### PR TITLE
KubeVirt Platform documentation for Advanced Storage Configurations and External Infrastructure

### PR DESCRIPTION
**Pull request by davidvossel**
This PR adds two new sections to the kubevirt how-to documentation
* storage configuration
* external infrastructure configuration

Now that the kubevirt documentation is growing larger, this PR also restructures some of the advanced networking configurations into a separate page in an effort to better organize the advanced content.

### CODE REVIEW

**Comment by davidvossel** [comment]
/cc @gqlo 

Maybe you can review this since you've been involved with the docs in the past?

File: docs/content/how-to/kubevirt/ingress-and-dns.md [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+
+## Default Ingress and DNS Behavior
+
+Every OpenShift cluster comes setup with a default application ingress
+controller which is expected have an wildcard DNS record associated with it.
+By default, guest clusters created using the Hypershift KubeVirt provider
+will automatically become a subdomain of the underlying OCP cluster that
+the KubeVirt VMs run on.
+
+For example, if an OCP cluster cluster has a default ingress DNS entry of
```
Comment (gqlo) [inline docs/content/how-to/kubevirt/ingress-and-dns.md:17]: `OCP cluster cluster`   typo :)

File: docs/content/how-to/kubevirt/ingress-and-dns.md [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+```
+
+In the next section we will fix that.
+
+### Step 2 - Set up the LoadBalancer
+
+
+!!! note
+
+    If your cluster is on bare-metal you may need MetalLB to be able to provision functional LoadBalancer services. Take a look at the section [Optional MetalLB Configuration Steps](#optional-metallb-configuration-steps).
```
Comment (gqlo) [inline docs/content/how-to/kubevirt/ingress-and-dns.md:95]: [Optional MetalLB Configuration Steps ] link does not seem to be working here.

File: docs/content/how-to/kubevirt/ingress-and-dns.md [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+For example, if an OCP cluster cluster has a default ingress DNS entry of
+`*.apps.mgmt-cluster.example.com`, then the default ingress of a KubeVirt
+guest cluster named `guest` running on that underlying OCP cluster will
+be `*.apps.guest.apps.mgmt-cluster.example.com`.
+
+!!! note
+
+    For this default ingress DNS to work properly, the underlying cluster
+    hosting the KubeVirt VMs must allow wildcard DNS routes. This can be
+    configured using the following cli command. ```oc patch ingresscontroller -n openshift-ingress-operator default --type=json -p '[{ "op": "add", "path": "/spec/routeAdmission", "value": {wildcardPolicy: "WildcardsAllowed"}}]'```
```
Comment (gqlo) [inline docs/content/how-to/kubevirt/ingress-and-dns.md:26]: Is there a way to make the patch command more visible here? It seems like ``` doesn't like `!!!note`

File: docs/content/how-to/kubevirt/configuring-storage.md [hunk 4]
```diff
@@ -0,0 +1,132 @@
+# Configuring Storage
+
+By default, if no advanced configuration is provided, the default storageclass
+is used for the the KubeVirt VM images, the kubevirt csi mapping, and the
```
[the remark this answers is not in the document]
Comment (davidvossel) [reply docs/content/how-to/kubevirt/configuring-storage.md:4]: my understanding is that the ovn db is going away, as a result of interconnect i believe?

File: docs/content/how-to/kubevirt/ingress-and-dns.md (hunk 2 above)
Comment (davidvossel) [reply docs/content/how-to/kubevirt/ingress-and-dns.md:95]: the same link seems to work when the docs are published. maybe they don't work when viewing in github as markdown though. 

https://hypershift-docs.netlify.app/how-to/kubevirt/create-kubevirt-cluster/#optional-metallb-configuration-steps

File: docs/content/how-to/kubevirt/ingress-and-dns.md (hunk 3 above)
Comment (davidvossel) [reply docs/content/how-to/kubevirt/ingress-and-dns.md:26]: it seems to display correctly when the docs are published, here's an example of how the note is interpreted today. https://hypershift-docs.netlify.app/how-to/kubevirt/create-kubevirt-cluster/#default-ingress-and-dns-behavior

File: docs/content/how-to/kubevirt/ingress-and-dns.md [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+By default, the HyperShift operator will configure the KubeVirt platform guest
+cluster's ingress and DNS behavior to reuse what is provided by the underlying
+infra cluster that the KubeVirt VMs are running on. This section describes
+that default behavior in greater detail as well as information on advanced usage
+options.
+
+## Default Ingress and DNS Behavior
+
+Every OpenShift cluster comes setup with a default application ingress
+controller which is expected have an wildcard DNS record associated with it.
```
[the remark this answers is not in the document]
Comment (davidvossel) [reply docs/content/how-to/kubevirt/ingress-and-dns.md]: fixed

File: docs/content/how-to/kubevirt/configuring-storage.md [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+--node-pool-replicas $WORKER_COUNT \
+--pull-secret $PULL_SECRET \
+--memory $MEM \
+--cores $CPU \
+--infra-storage-class-mapping=infra-sc1/guest-sc1 \
+--infra-storage-class-mapping=infra-sc2/guest-sc2
+```
+
+Once the guest cluster is created, the `guest-sc1` and `guest-sc2` storage
+classes will be visible within the guest cluster. When users creat a PVC within
```
[the remark this answers is not in the document]
Comment (davidvossel) [reply docs/content/how-to/kubevirt/configuring-storage.md]: fixed

File: docs/content/how-to/kubevirt/configuring-storage.md [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+## KubeVirt VM Root Volume Configuration
+
+The storage class used to host the KubeVirt Virtual Machine root volumes can be
+configured at cluster creation time using the `hypershift` cli tool and the
+`--root-volume-storage-class` argument. Likewise, the size of the volume can be
+configured using the `--root-volume-size` cli argument.
+
+Below is an example of setting a custom storage class and volume size for the
+KubeVirt VMs. The result will be a guest cluster with VMs hosted on 64Gi PVCs
+hosted by the ocs-storagecluster-ceph-rdb storage class
```
[the remark this answers is not in the document]
Comment (davidvossel) [reply docs/content/how-to/kubevirt/configuring-storage.md]: yep, fixed

File: docs/content/how-to/kubevirt/ingress-and-dns.md (hunk 1 above)
Comment (davidvossel) [reply docs/content/how-to/kubevirt/ingress-and-dns.md]: fixed

**Comment by davidvossel** [comment]
@gqlo i addressed your feedback. when you get a chance, ptal

File: docs/content/how-to/kubevirt/configuring-storage.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (gqlo) [reply docs/content/how-to/kubevirt/configuring-storage.md:4]: Aha, right. I didn't realize that.

**Review by gqlo — APPROVED** [review]
/LGTM

**Pull request merged**